src/util/types.h

```diff
@@ -3,8 +3,13 @@
 
 #include "util/math.h"
 
+#include <cstddef>
 #include <climits>
 
+// Signed integer type for all kinds of sizes, array indices and pointer
+// arithmetic. Its size (32-/64-bit) depends on the CPU architecture.
```

Comment:
// Signed integer type for POT array indices, sizes and pointer
// arithmetic. Its size (32-/64-bit) depends on the CPU architecture.
// This should be uses for all CSAMLE operations 
// since it is fast and allows compiler auto vectorizing. 
// For Qt container operations use just int 

"since it is fast and allows compiler auto vectorizing." verification pending